src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -0,0 +1,502 @@
+        }
+        return tags;
+    }
+
+    /**
+     * Creates a new {@link TemporaryPoint} at the given {@link Coordinate}. We can assume that any
+     * new Points will be created at the country boundaries, so we can add the
+     * {@link SyntheticBoundaryNodeTag} tag.
+     *
+     * @param coordinate
+     *            The {@link Coordinate} of the new point
+     * @param pointIdentifierFactory
+     *            The {@link CountrySlicingIdentifierFactory} to calculate new point identifier
+     * @param pointTags
+     *            The tags for this new point
+     * @return an {@link Optional} containing the {@link TemporaryPoint}
+     */
+    private Optional<TemporaryPoint> createNewPoint(final Coordinate coordinate,
+            final CountrySlicingIdentifierFactory pointIdentifierFactory,
+            final Map<String, String> pointTags)
+    {
+        if (!pointIdentifierFactory.hasMore())
+        {
+            logger.error(
+                    "Country Slicing exceeded maximum number of supported new points at Coordinate {}",
+                    AbstractIdentifierFactory.IDENTIFIER_SCALE, coordinate);
+            return Optional.empty();
+        }
+        else
+        {
+            // Add the synthetic boundary node tags
+            pointTags.put(SyntheticBoundaryNodeTag.KEY, SyntheticBoundaryNodeTag.YES.toString());
+
+            return Optional.of(new TemporaryPoint(pointIdentifierFactory.nextIdentifier(),
+                    new Location(Latitude.degrees(coordinate.y), Longitude.degrees(coordinate.x)),
+                    pointTags));
+        }
+    }
+
+    /**
+     * Assigns {@link ISOCountryTag}, {@link SyntheticNearestNeighborCountryCodeTag} and
+     * {@link SyntheticBoundaryNodeTag} values for a given {@link Location}.
+     *
+     * @param location
+     *            The {@link Location} for which to assign country code tags
+     * @param fromRawAtlas
+     *            {@code true} if this {@link Location} is already an {@link Point} in the raw Atlas
+     * @return the created tags
+     */
+    private Map<String, String> createPointTags(final Location location, final boolean fromRawAtlas)
+    {
+        final Map<String, String> tags = new HashMap<>();
+
+        // Get the country code details
+        final CountryCodeProperties countryDetails = this.countryBoundaryMap
+                .getCountryCodeISO3(location);
+
+        // Store the country code
+        tags.put(ISOCountryTag.KEY, countryDetails.getIso3CountryCode());
+
+        // If we used nearest neighbor logic to determine the country code, add a tag
+        // to indicate this
+        if (countryDetails.usingNearestNeighbor())
+        {
+            tags.put(SyntheticNearestNeighborCountryCodeTag.KEY,
+                    SyntheticNearestNeighborCountryCodeTag.YES.toString());
+        }
+
+        // For any border nodes, add the existing tag
+        if (fromRawAtlas && countryDetails.inMultipleCountries())
```

Comment:
Here does `fromRawAtlas` mean that the `Point` that generated this `Location` is already in the raw Atlas?

Reply:
The logic is that - if we're creating a new `Point`, it's only going to happen at the boundary, since that's where slicing happens. We have two ways of retrieving `Point`s - 1. from the original un-sliced Raw Atlas 2. from the `newPointCoordinates` map which maps a `Coordinate` to any `TemporaryPoint` identifier.

We first check to see if there is already a new `Point` for this `Coordinate` in the map (to avoid creating duplicate points). If there's not, we check the raw Atlas to see if there is already a `Point` at this location. If the atlas has a Point here, we know to add the `existing` boundary tag value, as opposed to the `yes` value. This has some corner cases, such as an existing node at the same coordinate that's not part of that line, but have yet to see this when testing.

Reply:
I'll add this as a comment

Reply:
Not sure this comment will help, but when two `Node`s are at the exact same location, the Atlas connects everything to one of the 2. The other node is still there, but is connected to nothing.